### ISSUE TITLE
Billing percentage should not show up in invoice create screen for which invoice is already created

### ISSUE THREAD

**Issue by rathorevaibhav** [ColoredCow/portal#460]
When we create a new invoice, the system shows all the billings for that client. Even the ones that are already attached to an invoice.

Ideally, the system should only show those billings who are not attached to any invoice.

**Comment by abhishek-pokhriyal** [comment]
From my verbal communication with @rathorevaibhav what I understood regarding this issue is:
  - We can create multiple bills for a stage of project.
  - We can then create invoice for each bill separately, right from the billings.
  - Once we create an invoice, the billing % for that bill gets locked, see picture below.
![screenshot 85](https://user-images.githubusercontent.com/11808845/41534813-87a9e604-731d-11e8-8749-fa73abe6417d.png)

  - But, we can also create an invoice for a billing by going through `Finance > Invoices > Create Invoice`
  - When creating an invoice from here (last step's location) a drop-down asks us to select the billing % for which we want to create an invoice.
  - **This drop-down should not give an option to select the billing % for which an invoice already exist.**
![screenshot 86](https://user-images.githubusercontent.com/11808845/41534995-41d3469c-731e-11e8-9c15-7ca350210410.png)

**Comment by abhishek-pokhriyal** [comment]
@rathorevaibhav this one requires some knowledge of VueJS. So it might take a week or so. Not sure about the estimate.

**Comment by rathorevaibhav** [comment]
@Satendra-SR assigning this to you. If @abhishek-pokhriyal prefers, you can take help from him on whatever he has researched in this.

**Comment by Satendra-SR** [comment]
#460
@rathorevaibhav 
I tried this one and I think it works fine.
but one thing I noticed that both **create invoice** and **edit invoice** are using same Vue template.
So how can we deal with this? We will not able to select **invoice %** that are already created in **edit invoice** also.

**Comment by rathorevaibhav** [comment]
@Satendra-SR I believe we're sorted on the areas of doubts based on our discussion. Just to summarize:

Consider the following billings for a project stage:
```
Billing%     InvoiceID
50%           1
20%           -
30%           -
```
There is an invoice for 50% billing and other billings don't have any invoice yet.

1. We need to put a check in the Vue template to see if the current page is `create` or `edit` invoice.
2. Next, in the create page, only 20% and 30% billings should show up in the select option (as they don't have any invoice).
3. In the edit page of `invoiceID 1`, 50% billing selection should show up by default. If the user adds a new project, then he should have ability to select only 20% and 30%.

**Comment by Satendra-SR** [comment]
I have completed the above. The system is only showing those billings who are not attached to an invoice.
but the system is also showing those stages which do not have any empty billing. it should not be shown up there. I tried it but it creates a bulky code and uses 3 for loops which is not a good practice as the code should be optimized.
I talked with @rathorevaibhav  and he suggested me for code refactoring. so I'm putting it on hold and will try to refactor it.
